File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <unordered_map>

using namespace std;

vector<int> solution(int n, vector<string> words) {
    vector<int> answer(2);
    unordered_map<string, int> check;
    int i = 1;
    check[words[0]] = 1;
    
    for (i; i < words.size(); i++)
    {
        if(check[words[i]] || words[i-1][words[i-1].size() - 1] != words[i][0]) // 앞단어의 끝 글자와 시작 글자가 같지 않을 경우
        {
            answer[0] = (i % n) + 1;
            answer[1] = (i / n) + 1;
            break;
        }
        check[words[i]] = 1;
    }
    if (i == words.size())
    {
        answer[0] = 0;
        answer[1] = 0;
    }

    return answer;
}

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    answer = []
    dic = dict()
    words.insert(0, words[0][0])
    for idx, value in enumerate(words):
        if len(value) == 1:
            continue
        if value[0] != words[idx-1][-1]:
            if idx % n == 0:
                return [n,idx//n]
            else:
                return [idx%n, idx//n+1]
        else:
            try:
                dic[value]
                if idx % n == 0:
                    return [n,idx//n]
                else:
                    return [idx%n, idx//n+1]    
            except:
                dic[value] = 1
    return [0, 0]